### PR TITLE
refactor tb_plugin code about trace event generation

### PR DESCRIPTION
**Pull request by guotuofeng**

### CODE REVIEW

File: tb_plugin/torch_tb_profiler/profiler/trace.py [hunk 1]
```diff
@@ -33,92 +34,32 @@ def __init__(self, type, data):
...
-                "Python": PythonEvent,
-            }
-        }
-
-    def _get_handler(self, type=None, category=None):
-        handlers = self._handlers.get(type, None)
-        if handlers is None:
+def create_event(event):
+    try:
+        type = event.get("ph")
```
Comment (guyang3532) [inline tb_plugin/torch_tb_profiler/profiler/trace.py:49]: type = event.get("ph", None) is unnecessary?

Comment (guotuofeng) [reply tb_plugin/torch_tb_profiler/profiler/trace.py:49]: From https://docs.python.org/3/library/stdtypes.html#dict.get, event.get("ph") is same with event.get("ph", None)

File: tb_plugin/torch_tb_profiler/profiler/run_generator.py [hunk 2]
```diff
@@ -2,9 +2,9 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # --------------------------------------------------------------------------
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
+from __future__ import absolute_import, division, print_function
+
+from torch_tb_profiler.profiler.overall_parser import ProfileRole
```
[the remark this answers is not in the document]
Comment (guotuofeng) [reply tb_plugin/torch_tb_profiler/profiler/run_generator.py:7]: will fix it

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py [hunk 3]
```diff
@@ -193,79 +230,53 @@ def fill_stats(node):
...
-                node.end_time = event.ts + event.duration
-                node.type = event.type
-                if "external id" in event.args:
-                    node.external_id = event.args["external id"]
-                elif "External id" in event.args:
-                    node.external_id = event.args["External id"]
-
-            corrid = event.args["correlation"] if "correlation" in event.args else None
-            input_shape = event.args["Input dims"] if "Input dims" in event.args else None
+            corrid = event.args.get("correlation", None)
```
Comment (guyang3532) [inline tb_plugin/torch_tb_profiler/profiler/module_parser.py:241]: corrid = event.args.get("correlation") ?

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py [hunk 4]
```diff
@@ -193,79 +230,53 @@ def fill_stats(node):
...
+            input_shape = event.args.get("Input dims", None)
```
Comment (guyang3532) [inline tb_plugin/torch_tb_profiler/profiler/module_parser.py:242]: input_shape = event.args.get("Input dims") ?

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py [hunk 5]
```diff
@@ -161,29 +214,13 @@ def remove_dup_nodes(node):
...
-                node.start_time = node.end_time = None
-                for i in range(len(node.children)):
-                    if node.children[i].start_time is not None:
-                        node.start_time = node.children[i].start_time
-                        break
-                for i in range(len(node.children) - 1, -1, -1):
-                    if node.children[i].end_time is not None:
-                        node.end_time = node.children[i].end_time
-                        break
+                    traverse_node(rt)
```
Comment (guyang3532) [inline tb_plugin/torch_tb_profiler/profiler/module_parser.py:223]: why delete the node.fill_stats() here ?

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py (hunk 4 above)
Comment (guotuofeng) [reply tb_plugin/torch_tb_profiler/profiler/module_parser.py:242]: yes, they are exactly same

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py (hunk 3 above)
Comment (guotuofeng) [reply tb_plugin/torch_tb_profiler/profiler/module_parser.py:241]: yes, they are exactly same

**Pull request merged**